Lib/Platform/POSIX/DiagnosticsPOSIX.cpp

```diff
@@ -70,19 +70,20 @@ bool Platform::getInstructionSourceByAddress(Uptr ip, InstructionSource& outSour
 		}
 		else
 		{
-			char demangledBuffer[1024];
+			char* demangledBuffer = nullptr; // will be allocated by __cxa_demangle
 			const char* demangledSymbolName = symbolInfo.dli_sname;
 			if(symbolInfo.dli_sname[0] == '_')
 			{
 				Uptr numDemangledChars = sizeof(demangledBuffer);
 				I32 demangleStatus = 0;
-				if(abi::__cxa_demangle(symbolInfo.dli_sname,
-									   demangledBuffer,
-									   (size_t*)&numDemangledChars,
-									   &demangleStatus))
-				{ demangledSymbolName = demangledBuffer; }
+				demangledBuffer = abi::__cxa_demangle(symbolInfo.dli_sname,
+								                      demangledBuffer,
+									                  (size_t*)&numDemangledChars,
+									                  &demangleStatus);
+				if(demangledBuffer) { demangledSymbolName = demangledBuffer; }
 			}
 			outSource.function = demangledSymbolName;
+			free(demangledBuffer); // was copied to outSource.function
 			outSource.instructionOffset = ip - reinterpret_cast<Uptr>(symbolInfo.dli_saddr);
```

Comment:
This whole `else` block can be replaced by something much simpler, i.e.
```
int status;
char* demangled = abi::__cxa_demangle(symbolInfo.dli_sname, nullptr, nullptr, &status);
if(demangled != nullptr)
{
    outSource.function = std::string(demangled);
    free(demangled);
}
else
{
    outSource.function = std::string(symbolInfo.dli_sname);
}
outSource.instructionOffset = ip - reinterpret_cast<Uptr>(symbolInfo.dli_saddr);
```

Reply:
std::string would throw exception. the code is not exception safe

Reply:
`outSource.function` is of type `std::string`, so `std::string` was already created implicitly during the assignment (I mean in this PR, not in the current version, which is broken for stated reasons).

If you're worried about the exception-safety, then `outSource.function` could be changed to use `char *` and `size_t` instead of `std::string`, but that's a bigger change, and I'm not sure if handling exceptions in `std::string()` is something that WAVM cares about.

@AndrewScheidecker any thoughts?